### ISSUE TITLE
I encountered an error installing mmf

### ISSUE THREAD

**Issue by qiaojiaxin** [facebookresearch/mmf#1307]
`Building wheels for collected packages: mmf, editdistance, fasttext
  Building editable for mmf (pyproject.toml) ... done
  Created wheel for mmf: filename=mmf-1.0.0rc12-0.editable-cp38-cp38-win_amd64.whl size=10907 sha256=076e3370866c5ddba75ff454adedcf0fccf2dd30f1ee187a257ee303d6556860
  Stored in directory: C:\Users\如晴天 似雨天\AppData\Local\Temp\pip-ephem-wheel-cache-8cprhw3t\wheels\1d\bd\09\3cbd09cc8743144ac6d82f5056a025e8517c849566aa171d81
  Building wheel for editdistance (setup.py) ... error
  error: subprocess-exited-with-error

  × python setup.py bdist_wheel did not run successfully.
  │ exit code: 1
  ╰─> [31 lines of output]
      running bdist_wheel
      running build
      running build_py
      creating build
      creating build\lib.win-amd64-cpython-38
      creating build\lib.win-amd64-cpython-38\editdistance
      copying editdistance\__init__.py -> build\lib.win-amd64-cpython-38\editdistance
      copying editdistance\_editdistance.h -> build\lib.win-amd64-cpython-38\editdistance
      copying editdistance\def.h -> build\lib.win-amd64-cpython-38\editdistance
      running build_ext
      building 'editdistance.bycython' extension
      creating build\temp.win-amd64-cpython-38
      creating build\temp.win-amd64-cpython-38\Release
      creating build\temp.win-amd64-cpython-38\Release\editdistance
      "C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\BIN\x86_amd64\cl.exe" /c /nologo /O2 /W3 /GL /DNDEBUG /MD -I./editdistance -IC:\ProgramData\anaconda3\include -IC:\ProgramData\anaconda3\Include "-IC:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\INCLUDE" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.10240.0\ucrt" "-IC:\Program Files (x86)\Windows Kits\8.1\include\shared" "-IC:\Program Files (x86)\Windows Kits\8.1\include\um" "-IC:\Program Files (x86)\Windows Kits\8.1\include\winrt" /EHsc /Tpeditdistance/_editdistance.cpp /Fobuild\temp.win-amd64-cpython-38\Release\editdistance/_editdistance.obj
      _editdistance.cpp
      editdistance/_editdistance.cpp: warning C4819: The file contains a character that cannot be represented in the current code page (936). Save the file in Unicode format to prevent data loss
      editdistance/_editdistance.cpp(117): error C2059: syntax error: 'if'
      editdistance/_editdistance.cpp(118): error C2059: syntax error: 'else'
      editdistance/_editdistance.cpp(119): error C2059: syntax error: 'else'
      editdistance/_editdistance.cpp(120): error C2059: syntax error: 'else'
      editdistance/_editdistance.cpp(121): error C2059: syntax error: 'else'
      editdistance/_editdistance.cpp(122): error C2059: syntax error: 'else'
      editdistance/_editdistance.cpp(123): error C2059: syntax error: 'else'
      editdistance/_editdistance.cpp(124): error C2059: syntax error: 'else'
      editdistance/_editdistance.cpp(125): error C2059: syntax error: 'else'
      editdistance/_editdistance.cpp(126): error C2059: syntax error: 'else'
      editdistance/_editdistance.cpp(127): error C2059: syntax error: 'return'
      editdistance/_editdistance.cpp(128): error C2059: syntax error: '}'
      editdistance/_editdistance.cpp(128): error C2143: syntax error: missing ';' before '}'
      error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio 14.0\\VC\\BIN\\x86_amd64\\cl.exe' failed with exit code 2
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for editdistance
  Running setup.py clean for editdistance
  Building wheel for fasttext (setup.py) ... error
  error: subprocess-exited-with-error

  × python setup.py bdist_wheel did not run successfully.
  │ exit code: 1
  ╰─> [30 lines of output]
      C:\Users\如晴天 似雨天\AppData\Roaming\Python\Python38\site-packages\setuptools\dist.py:788: UserWarning: Usage of dash-separated 'description-file' will not be supported in future versions. Please use the underscore name 'description_file' instead
        warnings.warn(
      running bdist_wheel
      running build
      running build_py
      creating build
      creating build\lib.win-amd64-cpython-38
      creating build\lib.win-amd64-cpython-38\fasttext
      copying python\fasttext_module\fasttext\FastText.py -> build\lib.win-amd64-cpython-38\fasttext
      copying python\fasttext_module\fasttext\__init__.py -> build\lib.win-amd64-cpython-38\fasttext
      creating build\lib.win-amd64-cpython-38\fasttext\util
      copying python\fasttext_module\fasttext\util\util.py -> build\lib.win-amd64-cpython-38\fasttext\util
      copying python\fasttext_module\fasttext\util\__init__.py -> build\lib.win-amd64-cpython-38\fasttext\util
      creating build\lib.win-amd64-cpython-38\fasttext\tests
      copying python\fasttext_module\fasttext\tests\test_configurations.py -> build\lib.win-amd64-cpython-38\fasttext\tests
      copying python\fasttext_module\fasttext\tests\test_script.py -> build\lib.win-amd64-cpython-38\fasttext\tests
      copying python\fasttext_module\fasttext\tests\__init__.py -> build\lib.win-amd64-cpython-38\fasttext\tests
      running build_ext
      building 'fasttext_pybind' extension
      creating build\temp.win-amd64-cpython-38
      creating build\temp.win-amd64-cpython-38\Release
      creating build\temp.win-amd64-cpython-38\Release\python
      creating build\temp.win-amd64-cpython-38\Release\python\fasttext_module
      creating build\temp.win-amd64-cpython-38\Release\python\fasttext_module\fasttext
      creating build\temp.win-amd64-cpython-38\Release\python\fasttext_module\fasttext\pybind
      creating build\temp.win-amd64-cpython-38\Release\src
      "C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\BIN\x86_amd64\cl.exe" /c /nologo /O2 /W3 /GL /DNDEBUG /MD "-IC:\Users\如晴天 似雨天\AppData\Roaming\Python\Python38\site-packages\pybind11\include" "-IC:\Users\如晴天 似雨天\AppData\Roaming\Python\Python38\site-packages\pybind11\include" -Isrc -IC:\ProgramData\anaconda3\include -IC:\ProgramData\anaconda3\Include "-IC:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\INCLUDE" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.10240.0\ucrt" "-IC:\Program Files (x86)\Windows Kits\8.1\include\shared" "-IC:\Program Files (x86)\Windows Kits\8.1\include\um" "-IC:\Program Files (x86)\Windows Kits\8.1\include\winrt" /EHsc /Tppython/fasttext_module/fasttext/pybind/fasttext_pybind.cc /Fobuild\temp.win-amd64-cpython-38\Release\python/fasttext_module/fasttext/pybind/fasttext_pybind.obj /EHsc /DVERSION_INFO=\\\"0.9.1\\\"
      fasttext_pybind.cc
      c:\users\如晴天 似雨天\appdata\roaming\python\python38\site-packages\pybind11\include\pybind11\detail/common.h(147): fatal error C1189: #error:  pybind11 2.10+ requires MSVC 2017 or newer
      error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio 14.0\\VC\\BIN\\x86_amd64\\cl.exe' failed with exit code 2
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for fasttext
  Running setup.py clean for fasttext Successfully built mmf Failed to build editdistance fasttext Installing collected packages: editdistance, chardet, antlr4-python3-runtime, xxhash, urllib3, tzdata, tqdm, torch, smmap, pyparsing, pyDeprecate, portalocker, pillow, omegaconf, numpy, multidict, kiwisolver, idna, ftfy, fsspec, frozenlist, dill, async-timeout, yarl, torchmetrics, torchaudio, sacremoses, requests, pyarrow, pandas, nltk, multiprocess, matplotlib, iopath, gitdb, fasttext, aiosignal, torchvision, torchtext, scikit-learn, pycocotools, huggingface-hub, GitPython, datasets, aiohttp, transformers, sklearn, pytorch-lightning, mmf
  Attempting uninstall: editdistance
    Found existing installation: editdistance 0.6.2
    Uninstalling editdistance-0.6.2:
      Successfully uninstalled editdistance-0.6.2
  Running setup.py install for editdistance ... error
  error: subprocess-exited-with-error

  × Running setup.py install for editdistance did not run successfully.
  │ exit code: 1
  ╰─> [33 lines of output]
      running install
      C:\Users\如晴天 似雨天\AppData\Roaming\Python\Python38\site-packages\setuptools\command\install.py:34: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
        warnings.warn(
      running build
      running build_py
      creating build
      creating build\lib.win-amd64-cpython-38
      creating build\lib.win-amd64-cpython-38\editdistance
      copying editdistance\__init__.py -> build\lib.win-amd64-cpython-38\editdistance
      copying editdistance\_editdistance.h -> build\lib.win-amd64-cpython-38\editdistance
      copying editdistance\def.h -> build\lib.win-amd64-cpython-38\editdistance
      running build_ext
      building 'editdistance.bycython' extension
      creating build\temp.win-amd64-cpython-38
      creating build\temp.win-amd64-cpython-38\Release
      creating build\temp.win-amd64-cpython-38\Release\editdistance
      "C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\BIN\x86_amd64\cl.exe" /c /nologo /O2 /W3 /GL /DNDEBUG /MD -I./editdistance -IC:\ProgramData\anaconda3\include -IC:\ProgramData\anaconda3\Include "-IC:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\INCLUDE" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.10240.0\ucrt" "-IC:\Program Files (x86)\Windows Kits\8.1\include\shared" "-IC:\Program Files (x86)\Windows Kits\8.1\include\um" "-IC:\Program Files (x86)\Windows Kits\8.1\include\winrt" /EHsc /Tpeditdistance/_editdistance.cpp /Fobuild\temp.win-amd64-cpython-38\Release\editdistance/_editdistance.obj
      _editdistance.cpp
      editdistance/_editdistance.cpp: warning C4819: The file contains a character that cannot be represented in the current code page (936). Save the file in Unicode format to prevent data loss
      editdistance/_editdistance.cpp(117): error C2059: syntax error: 'if'
      editdistance/_editdistance.cpp(118): error C2059: syntax error: 'else'
      editdistance/_editdistance.cpp(119): error C2059: syntax error: 'else'
      editdistance/_editdistance.cpp(120): error C2059: syntax error: 'else'
      editdistance/_editdistance.cpp(121): error C2059: syntax error: 'else'
      editdistance/_editdistance.cpp(122): error C2059: syntax error: 'else'
      editdistance/_editdistance.cpp(123): error C2059: syntax error: 'else'
      editdistance/_editdistance.cpp(124): error C2059: syntax error: 'else'
      editdistance/_editdistance.cpp(125): error C2059: syntax error: 'else'
      editdistance/_editdistance.cpp(126): error C2059: syntax error: 'else'
      editdistance/_editdistance.cpp(127): error C2059: syntax error: 'return'
      editdistance/_editdistance.cpp(128): error C2059: syntax error: '}'
      editdistance/_editdistance.cpp(128): error C2143: syntax error: missing ';' before '}'
      error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio 14.0\\VC\\BIN\\x86_amd64\\cl.exe' failed with exit code 2
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  Rolling back uninstall of editdistance
  Moving to c:\users\如晴天 似雨天\appdata\roaming\python\python38\site-packages\editdistance-0.6.2.dist-info\
   from C:\Users\如晴天 似雨天\AppData\Roaming\Python\Python38\site-packages\~ditdistance-0.6.2.dist-info
  Moving to c:\users\如晴天 似雨天\appdata\roaming\python\python38\site-packages\editdistance\
   from C:\Users\如晴天 似雨天\AppData\Roaming\Python\Python38\site-packages\~ditdistance error: legacy-install-failure

× Encountered error while trying to install package.
╰─> editdistance

note: This is an issue with the package mentioned above, not pip.
hint: See above for output from the failure.`

**Comment by soumyasj** [comment]
Even I am facing the same issue.
